src/IO/Observer/Actions/ObserverRegistration.hpp

```diff
@@ -200,6 +315,70 @@ struct RegisterReductionContributorWithObserverWriter {
   }
 };
 
+struct DeregisterReductionContributorWithObserverWriter {
+ public:
+  template <typename ParallelComponent, typename DbTagsList,
+            typename Metavariables, typename ArrayIndex>
+  static void apply(db::DataBox<DbTagsList>& box,
+                    Parallel::GlobalCache<Metavariables>& cache,
+                    const ArrayIndex& /*array_index*/,
+                    const observers::ObservationKey& observation_key,
+                    const ArrayComponentId& id_of_caller) noexcept {
+    if constexpr (tmpl::list_contains_v<
+                      DbTagsList, Tags::ExpectedContributorsForObservations>) {
+      auto& my_proxy =
+          Parallel::get_parallel_component<ParallelComponent>(cache);
+      const auto node_id =
+          static_cast<size_t>(Parallel::my_node(*my_proxy.ckLocalBranch()));
+      db::mutate<Tags::ExpectedContributorsForObservations>(
+          make_not_null(&box),
+          [&cache, &id_of_caller, &node_id, &observation_key](
+              const gsl::not_null<std::unordered_map<
+                  ObservationKey, std::unordered_set<ArrayComponentId>>*>
+                  reduction_observers_registered) noexcept {
+            if (UNLIKELY(
+                    reduction_observers_registered->find(observation_key) ==
+                    reduction_observers_registered->end())) {
+              ERROR(
+                  "Trying to deregister a component associated with an "
+                  "unregistered observation key"
+                  << observation_key);
+            }
+
+            if (LIKELY(reduction_observers_registered->at(observation_key)
```

Comment:
Swap the if and else blocks for consistency with the rest of your error checks.  (And then the else indentation isn't needed.)